casual/casual-jca/src/test/groovy/se/laz/casual/jca/pool/NetworkConnectionPoolTest.groovy

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2022, The casual project. All rights reserved.
+ *
+ * This software is licensed under the MIT license, https://opensource.org/licenses/MIT
+ */
+package se.laz.casual.jca.pool
+
+import se.laz.casual.internal.network.NetworkConnection
+import se.laz.casual.jca.Address
+import se.laz.casual.jca.CasualResourceAdapterException
+import se.laz.casual.network.ProtocolVersion
+import se.laz.casual.network.outbound.NetworkListener
+import spock.lang.Specification
+
+class NetworkConnectionPoolTest extends Specification
+{
+   def 'using the wrong address'()
+   {
+      given:
+      int poolSize = 5
+      def poolName = 'small-pool'
+      Address address = Address.of("nifty", 7771)
+      Address anotherAddress = Address.of('delta', 8787)
+      NetworkConnectionCreator connectionCreator = Mock(NetworkConnectionCreator){
+         1 * createNetworkConnection(*_) >> Mock(ReferenceCountedNetworkConnection)
+      }
+      NetworkConnectionPool pool = NetworkConnectionPool.of(poolName, address, poolSize, connectionCreator)
+      ProtocolVersion protocolVersion = ProtocolVersion.VERSION_1_0
+      when: // working as expected with the correct address
+      NetworkConnection connection = pool.getOrCreateConnection(address, protocolVersion, Mock(NetworkListener))
+      then:
+      noExceptionThrown()
+      connection != null
+      when: // using the wrong address - throws
+      pool.getOrCreateConnection(anotherAddress, protocolVersion, Mock(NetworkListener))
+      then:
+      thrown(CasualResourceAdapterException)
+   }
+
+   def 'size 1 always returns the same instance'()
+   {
+      int poolSize = 1
```

Comment:
given? setup?
Also what do we say about new lines before each block; when / then etc. We used to do this and it's much more readable in my opinion.

Reply:
yeah, that should be there for sure.
Didn't even know you could write spock tests without given or setup.

Reply:
Fixed